### PR TITLE
[RNMobile] Migrate native TS files to JS

### PR DESCRIPTION
**Pull request by fluiddot**
Related to https://github.com/wordpress-mobile/gutenberg-mobile/issues/5622.

## Proposed changes:

* https://github.com/Automattic/jetpack/pull/29894/commits/416f80070c29b97d99c8b7c1b126868e6da6e8da: All files have been renamed without any modifications.
* https://github.com/Automattic/jetpack/pull/29894/commits/f24b6cbf8a5082f47085684a33df1985970fd1d8, https://github.com/Automattic/jetpack/pull/29894/commits/b189b2ceb09caeec9f65f64c160d776c3f263600, https://github.com/Automattic/jetpack/pull/29894/commits/68ad35486bbf4505e659375cba2c0e924afa39ca, and https://github.com/Automattic/jetpack/pull/29894/commits/376111e4b1a4fe89f8efbd55895e3819a8a2ea6c: The different native TS files have been updated to remove TypeScript.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

These changes shouldn't introduce side effects, but since several files have been updated, it would be great to test the user flows related to those files just in case.

### Block settings
Follow testing instructions from:
* https://github.com/Automattic/jetpack/pull/29283
* https://github.com/Automattic/jetpack/pull/29457
* https://github.com/Automattic/jetpack/pull/29567
* https://github.com/Automattic/jetpack/pull/29477

### Player
Follow testing instructions from https://github.com/Automattic/jetpack/pull/29478.

### Replace control
Follow testing instructions from https://github.com/Automattic/jetpack/pull/29662.

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js:1]: I created the native version of this file as the original is a TypeScript file. Eventually, we could move this code to the `index` file as this component was created to reuse the logic between platforms, but due to this change, this is no longer the case.

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.js [hunk 1]
```diff
@@ -11,40 +11,38 @@ import { createBlock } from '@wordpress/blocks';
 import { PanelBody } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { useState, useCallback, useEffect } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.js:14]: I noticed `sprintf ` function wasn't being imported in the original code.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		label: _x( 'Public', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+	};
+	const privacyOptionPrivate = {
+		value: String( VIDEO_PRIVACY_LEVELS.indexOf( VIDEO_PRIVACY_LEVEL_PRIVATE ) ),
+		label: _x( 'Private', 'VideoPress privacy setting', 'jetpack-videopress-pkg' ),
+	};
+
+	return (
+		<PanelBody
+			title={ Platform.isWeb && __( 'Privacy and rating', 'jetpack-videopress-pkg' ) }
```
Comment (SiobhyB) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js]: We won't need this check in either the web or native files following this change. 

In addition, this separate `<PrivacyAndRatingSettings/>` component was created in https://github.com/Automattic/jetpack/pull/29477/commits/43b954ff643a88b63980780bc31b337fc9f1823f in order to share as much functionality between the web and native as possible. As the reasoning behind that commit is now somewhat redundant, what do you think about reverting back to having all of the code for the Privacy and Rating component in the `index` files? I could see a case for keeping this more modular and don't feel overly strongly, but also don't think there's a strong need for this to be a separate component now. Happy to go with whatever your preference is here.

Comment (fluiddot) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js]: > We won't need this check in either the web or native files following this change.

Good catch, I'll remove it for the native file.

> In addition, this separate <PrivacyAndRatingSettings/> component was created in https://github.com/Automattic/jetpack/commit/43b954ff643a88b63980780bc31b337fc9f1823f in order to share as much functionality between the web and native as possible. As the reasoning behind that commit is now somewhat redundant, what do you think about reverting back to having all of the code for the Privacy and Rating component in the index files? I could see a case for keeping this more modular and don't feel overly strongly, but also don't think there's a strong need for this to be a separate component now. Happy to go with whatever your preference is here.

@SiobhyB I totally agree, now that we need to separate `PrivacyAndRatingSettings` component into two files we can revert the change introduced in https://github.com/Automattic/jetpack/commit/43b954ff643a88b63980780bc31b337fc9f1823f. This PR is subject to get file conflicts as it's updating most of the native files, so if possible, I'd like to propose merging it as is and applying the revert in a separate PR. WDYT?

Comment (fluiddot) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js]: > > We won't need this check in either the web or native files following this change.
> 
> Good catch, I'll remove it for the native file.

Fix applied in https://github.com/Automattic/jetpack/pull/29894/commits/e518e8a877d9a4cc6651939219a5fc40caa9639b.

Comment (SiobhyB) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/privacy-and-rating-settings.native.js]: > I'd like to propose merging it as is and applying the revert in a separate PR. WDYT?

@fluiddot, that makes sense to me, and apologies I'd originally missed [your comment](https://github.com/Automattic/jetpack/pull/29894/commits/e518e8a877d9a4cc6651939219a5fc40caa9639b#r1157005332) stating that intention. Thanks for removing the platform-specific condition, too!

**Review by SiobhyB — APPROVED** [review]
LGTM, thanks @fluiddot 🙌

**Review by dhasilva — APPROVED** [review]
No regressions on web version :+1:

**Pull request merged**